### PR TITLE
feat: add broker token

### PR DESCRIPTION
**Pull request by maxpushka**
- **feat: add BrokerToken contract**
- **forge install: openzeppelin-contracts**
- **build: add openzeppelin/contracts as a dependency**
- **forge install: openzeppelin-contracts**

### CODE REVIEW

File: foundry.toml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 [profile.default]
-src = "src"
-out = "out"
-libs = ["lib"]
-
-# See more config options https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
+src = 'src'
+out = 'out'
+libs = ['node_modules', 'lib']
```
Comment (mod) [inline foundry.toml]: We should not have node_modules

File: src/BrokerToken.sol [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	 * @param name Name of the Token.
+	 * @param symbol Symbol of the Token.
+	 * @param supply Maximum supply of the Token.
+	 */
+	constructor(
+		string memory name,
+		string memory symbol,
+		uint256 supply
+	) ERC20Permit(name) ERC20(name, symbol) {
+		_mint(msg.sender, supply);
```
Comment (mod) [inline src/BrokerToken.sol]: Maybe we need to do supply x 10^decimals ?
or mint does it ?

Comment (maxpushka) [reply src/BrokerToken.sol]: No, `_mint` does not do this for us. Usually the user does the math himself

File: foundry.toml (hunk 1 above)
Comment (maxpushka) [reply foundry.toml]: da9f787

**Review by mod — APPROVED** [review]

**Pull request merged**